lib/phase/functional.js

```diff
@@ -91,6 +92,26 @@ function validateBuildClusterAnnotation(doc, buildClusterFactory) {
     return Promise.resolve([]);
 }
 
+/**
+ * Ensure the triggers for a job are valid
+ * @method isJobTriggersValid
+ * @param  {Array}  triggers    List of requires configured for a job
+ * @param  {Array}  options     List of valid options to trigger a job
+ * @return {Boolean}
+ *
+ */
+function isJobTriggersValid(triggers, options) {
```

Comment:
Let us keep it specific. For non-stage jobs we may never need such custom checks.
```suggestion
function isStageJobTriggersValid(jobTriggers, stageJobNames) {
```

Reply:
Validator does not produce an error for a normal workflow for a job trigger that does not exist. So maybe this is on purpose because sometimes trigger comes from an external pipeline?

![Screenshot 2024-05-13 at 10 41 18 AM](https://github.com/screwdriver-cd/config-parser/assets/7771180/6243fad4-82e3-4e9a-bce2-5b248a68e252)